staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -303,6 +339,9 @@ func (mounter *Mounter) Unmount(target string) error {
 			// Rewrite err with the actual exit error of the process.
 			err = &exec.ExitError{ProcessState: command.ProcessState}
 		}
+		if mounter.withSafeNotMountedBehavior && strings.Contains(string(output), errNotMounted) {
```

Comment:
is this in the right place? if umount exits 1 I think err will be non-nil so this belongs in the above if statement. Will need to test.

1. create Pod with a volume
2. ssh onto Node and unmount one of its volumes
3. delete Pod
4. does this condition actually get hit?

Reply:
This is within the `if err != nil {}` block; and I think it should come *after* the `ErrNoChildProcess` check (as is). I'll do that test, to confirm.

Reply:
put a log here, and update Unmount comment. 

Since we are changing the library here, we also need to make sure this behavior change will not affect any other logic.

Reply:
many in-tree drivers uses this function, could you double check this behavior change will not affect any of them?

Reply:
From a quick search, I think corrupted mounts often result in `failed: Operation not permitted` from `umount`. That behavior won't change with this PR, the error will be returned.